File: lib-segment/src/main/java/websiteschema/mpsegment/dict/UnknownWord.java
package websiteschema.mpsegment.dict;

import websiteschema.mpsegment.concept.Concept;

/**
 *
 * @author ray
 */
public class UnknownWord implements IWord {

    final static int log2Freq = (int) (Math.log(1) * 100D);
    String wordName;

    public UnknownWord(String wordName) {
        this.wordName = wordName;
    }

    @Override
    public int getLog2Freq() {
        return log2Freq;
    }

    @Override
    public int getDomainType() {
        return 0;
    }

    @Override
    public long getOccuredCount(String s) {
        if (s.equals(POSUtil.getPOSString(POSUtil.POS_UNKOWN))) {
            return 1L;
        }
        return 0L;
    }

    @Override
    public long getOccuredSum() {
        return 1L;
    }

    @Override
    public POSArray getPOSArray() {
        return POSArrayFactory.getUnknownWordPOSArray();
    }

    @Override
    public int[][] getWordPOSTable() {
        return getPOSArray().getWordPOSTable();
    }

    @Override
    public int getWordLength() {
        return wordName.length();
    }

    @Override
    public String getWordName() {
        return wordName;
    }

    @Override
    public void incOccuredCount(String s) {
    }

    @Override
    public void setDomainType(int i) {
    }

    @Override
    public void setOccuredCount(String s, int i) {
    }

    @Override
    public void setOccuredSum(int i) {
    }

    @Override
    public int getWordMaxPOS() {
        return POSUtil.POS_UNKOWN;
    }

    @Override
    public String toDBFString() {
        StringBuilder stringbuffer = new StringBuilder();
        stringbuffer.append(getPOSArray().toDBFString(getWordName()));
        return stringbuffer.toString();
    }

    @Override
    public Concept[] getConcepts() {
        return new Concept[]{Concept.UNKNOWN};
    }

}

File: lib-segment-scala/src/main/scala/websiteschema/mpsegment/dict/UnknownWord.scala
package websiteschema.mpsegment.dict;

import websiteschema.mpsegment.concept.Concept;

object UnknownWord {

  def apply(wordName: String) = {
    val word = new UnknownWord()
    word.wordName = wordName
    word
  }
}

class UnknownWord extends IWord {

    var wordName: String = ""

    override def getLog2Freq() : Int = {
        return 0;
    }

    override def getDomainType() : Int = {
        return 0;
    }

    override def getOccuredCount(s: String) : Long = {
        if (s.equals(POSUtil.getPOSString(POSUtil.POS_UNKOWN))) {
            return 1L;
        }
        return 0L;
    }

    override def getOccuredSum() : Long = {
        return 1L;
    }

    override def getPOSArray() : POSArray = {
        return POSArrayFactory.getUnknownWordPOSArray();
    }

    override def getWordPOSTable() = getPOSArray().getWordPOSTable()

    override def getWordLength() : Int = {
        return wordName.length();
    }

    override def getWordName() : String = {
        return wordName;
    }

    override def incOccuredCount(s: String) {
    }

    override def setDomainType(i: Int) {
    }

    override def setOccuredCount(s: String, i: Int) {
    }

    override def setOccuredSum(i: Int) {
    }

    override def getWordMaxPOS() : Int = {
        return POSUtil.POS_UNKOWN;
    }

    override def toDBFString() : String = {
        val stringBuilder = new StringBuilder()
        stringBuilder.append(getPOSArray().toDBFString(getWordName()));
        return stringBuilder.toString();
    }

    override def getConcepts() : Array[Concept] = {
      val array = new Array[Concept](1)
      array(0) = Concept.UNKNOWN
      array
    }

}
